### PR TITLE
[FIX] Return of Purchase order

### PR DESCRIPTION
**Pull request by TimLai125**

### CODE REVIEW

**Comment by TimLai125** [comment]
The PR covers when the the delivery order of a purchase order is being returned/
There will be no sale order for this case. hence the following line will generate error. 
`so_rec = self.env['sale.order'].search([('procurement_group_id', '=', self.group_id.id)])[0]`

File: stock_delivery_notification/models/stock_picking.py [hunk 1]
```diff
@@ -35,13 +35,13 @@ def action_send_delivery_order(self):
         self.ensure_one()
         if self.picking_type_id.code == "outgoing" and self.group_id:
```
Comment (yostashiro) [inline stock_delivery_notification/models/stock_picking.py]:
```
        if self.picking_type_id.code == "outgoing" and self.sale_id and self.group_id:
```

Haven't tested, but I'd personally prefer this over your suggestion as it'd make the context clearer.

**Review by yostashiro — COMMENTED** [review]
Thanks for looking into this!

File: stock_delivery_notification/models/stock_picking.py (hunk 1 above)
Comment (yostashiro) [reply stock_delivery_notification/models/stock_picking.py]: I think we can further simplify the code using self.sale_id, actually (our code is redundant with Odoo's standard logic).

See how sale_id is computed here: https://github.com/odoo/odoo/blob/6fa705399bc05d4b0dce5e7181c40a4a5ab8b1c3/addons/sale_stock/models/stock.py#L51-L55

**Comment by TimLai125** [comment]
The method is simplified by using self.sale_id

**Review by yostashiro — APPROVED** [review]
I'd personally even remove `sale_order = self.sale_id` but LGTM.

**Pull request merged**